core/src/main/java/org/apache/gravitino/stats/storage/PartitionStatisticStorage.java

```diff
@@ -0,0 +1,119 @@
+        "Don't support listStatistics with partition names yet.");
+  }
+
+  /**
+   * Appends statistics to the storage for a given metadata object. The implementation should
+   * guarantee the thread safe. The upper layer will add the parent object level write lock. For
+   * example, if the metadata object is a table, the write lock of the schema level will be held.
+   *
+   * @param metalake the name of the metalake
+   * @param statisticsToAppend a list of {@link MetadataObjectStatisticsUpdate} objects, each
+   *     containing the metadata object and its associated statistics updates.
+   */
+  void appendStatistics(String metalake, List<MetadataObjectStatisticsUpdate> statisticsToAppend);
+
+  /**
+   * Drops statistics for specified partitions of a metadata object. The implementation should
+   * guarantee the thread safe. The upper layer will add the parent object level write lock. For
+   * example, if the metadata object is a table, the write lock of the schema level will be held.
+   *
+   * @param metalake the name of the metalake
+   * @param partitionStatisticsToDrop a map where the key is a {@link MetadataObject} and the value
+   *     is a list of {@link PartitionStatisticsDrop}
+   */
+  void dropStatistics(
+      String metalake, List<MetadataObjectStatisticsDrop> partitionStatisticsToDrop);
+
+  /**
+   * Updates statistics for a given metadata object. The default implementation is to first drop and
+   * then append the statistics. Developer can override this logic if needed. The implementation
+   * should guarantee the thread safe. The upper layer will add the parent object level write lock.
+   * For example, if the metadata object is a table, the write lock of the schema level will be
+   * held.
+   *
+   * @param metalake the name of the metalake
+   * @param statisticsToUpdate a list of {@link MetadataObjectStatisticsUpdate} objects, each
+   *     containing the metadata object and its associated statistics updates.
+   */
+  default void updateStatistics(
+      String metalake, List<MetadataObjectStatisticsUpdate> statisticsToUpdate) {
+    List<MetadataObjectStatisticsDrop> statisticsToDrop =
+        statisticsToUpdate.stream()
+            .map(
+                update ->
+                    MetadataObjectStatisticsDrop.of(
+                        update.metadataObject(),
+                        update.partitionUpdates().stream()
+                            .map(
+                                partitionUpdate ->
+                                    PartitionStatisticsDrop.of(
+                                        partitionUpdate.partitionName(),
+                                        Lists.newArrayList(partitionUpdate.statistics().keySet())))
+                            .collect(Collectors.toList())))
+            .collect(Collectors.toList());
+
+    dropStatistics(metalake, statisticsToDrop);
+    appendStatistics(metalake, statisticsToUpdate);
+  }
+}
```

Comment:
Why can we provide a "update" interface, let user to decide how to implement the update semantic, I think here the default operation is low-efficient.

Again, why do we provide a "append" API, and use "drop" + "append" to achieve "update"? We can always use "update" to achieve upsert semantic.

Reply:
What if a partial statistics update fails?

Reply:
It depends on the underlying storage system.